src/views/v2/commerce/checkout/purchaser-info.php

```diff
@@ -18,13 +18,45 @@
  * @var bool $must_login Global] Whether login is required to buy tickets or not.
  */
 
-if ( is_user_logged_in() || $must_login || empty( $items ) ) {
+// Bail if the cart is empty.
+if ( empty( $items ) ) {
+	return;
+}
+
+// Bail if user needs to login, but is not logged in.
+if ( ! is_user_logged_in() && $must_login ) {
 	return;
 }
-?>
 
+$info_title   = __( 'Purchaser info', 'event-tickets' );
+$show_address = false;
+foreach ( $gateways as $gateway ) {
+	// Check if Stripe is active and enabled.
+	if ( 'stripe' === $gateway::get_key() && $gateway::is_enabled() && $gateway::is_active() ) {
```

Comment:
Our ~standard~ general practice is to split to multi-lines when the conditionals exceed two for readability:

_Note: I thought I had put that in our standards, but I guess that is something that we don't draw a hard line on, just encourage for readability._

```suggestion
	if (
		'stripe' === $gateway::get_key()
		&& $gateway::is_enabled()
		&& $gateway::is_active()
	) {
```